hack/test-cmd.sh

```diff
@@ -57,7 +57,7 @@ tests=( $(find_tests ${1:-.*}) )
 
 # test-cmd specific defaults
 TMPDIR="${TMPDIR:-"/tmp"}"
-BASETMPDIR="${BASETMPDIR:-${TMPDIR}/openshift-cmd}"
+export BASETMPDIR="${BASETMPDIR:-${TMPDIR}/openshift-cmd}"
```

Comment:
i'll change it to the unit test approach, but the claim that diskIO is a potential issue is kinda silly, all our tests depend on diskIO.  I don't see anything about this approach that's likely to be particularly flaky.

and cleanup is handled in the same way the log files are cleaned up, because it's writing to the same location.

I think this is still a simpler approach than the equivalent go code, even if it's not simpler than the other bash tests in this file.